alf/utils/common.py

```diff
@@ -440,23 +453,28 @@ def _traverse_path(obs, path):
         if not path:
             return _transform_image(obs)
         step = path[0]
-        if isinstance(obs, tuple) and hasattr(obs, '_fields'):
-            new_val = _traverse_path(getattr(obs, step), path[1:])
-            return obs._replace(**{step: new_val})
+        if isinstance(obs, tuple):
+            if hasattr(obs, '_fields'):
+                new_val = _traverse_path(getattr(obs, step), path[1:])
+                return obs._replace(**{step: new_val})
+            else:
+                step = int(step)
+                transformed = _traverse_path(obs[step], path[1:])
+                return obs[:step] + (transformed, ) + obs[step + 1:]
         elif isinstance(obs, dict):
             new_obs = obs.copy()
             new_obs[step] = _traverse_path(obs[step], path[1:])
             return new_obs
         else:
             raise TypeError(("If observation is a nest, it must be either " +
-                             "a dict or namedtuple!"))
+                             "a dict or tuple!"))
 
     fields = fields or [""]
     for field in fields:
         # remove '' in the path
         path = [step for step in field.split(".") if step]
         observation = _traverse_path(observation, path)
-    return observation
+    return cast_transformer(observation, tf.float32)
```

Comment:
This will fail if observation has language input

Reply:
removed